modules/link.py

```diff
@@ -29,12 +29,31 @@ def __init__(self, link, *, tld=False):
         else:
             self.name = self._node.title.string
             self.status = color(link, 'green')
-
+    
+    def get_children(self):       
+        if self._children:
+            return self._children
+        
+        self._children = self._node.find_all('a')
+        return self._node.find_all('a')
+       
+    def get_urls(self):
+        if self._urls:
+            return self._urls
+        
+        children = get_children()
+        url_nodes = []
+        for child in children:
+            link = child.get('href')
+            url_nodes.append(link) if valid_link(link)
```

Comment:
This should probably be changed to
```suggestion
            if self.valid_link(link) and link:
                link_nodes.append(link)
```